tests/integration/Settings_API/AbstractSettingsTest.php

```diff
@@ -86,6 +87,76 @@ public function test_unregister_setting() {
 	}
 
 
+	/**
+	 * @see Abstract_Settings::registr_control()
+	 *
+	 * @param string $setting_id the setting ID
+	 * @param string $control_type the control type
+	 * @param bool $registered whether the control should be succesfully registered or not
+	 *
+	 * @dataProvider provider_register_control
+	 */
+	public function test_register_control( $setting_id, $control_type, $registered ) {
+
+		$this->get_settings_instance()->register_setting( 'registered_setting', Setting::TYPE_STRING );
+
+		$this->assertSame( $registered, $this->get_settings_instance()->register_control( $setting_id, $control_type ) );
+
+		if ( $registered ) {
+
+			$this->assertInstanceOf( Control::class, $this->get_settings_instance()->get_setting( $setting_id )->get_control() );
+
+		} elseif ( $setting = $this->get_settings_instance()->get_setting( $setting_id ) ) {
+
+			$this->assertNull( $setting->get_control() );
+		}
+	}
+
+
+	/** @see test_register_control() */
+	public function provider_register_control() {
+
+		require_once 'woocommerce/Settings_API/Control.php';
+
+		return [
+			[ 'unknown_setting',    Control::TYPE_TEXT, false ],
+			[ 'registered_setting', 'invalid_type',     false ],
+			[ 'registered_setting', Control::TYPE_TEXT, true ],
+		];
+	}
+
+
+	/** @see Abstract_Settings::register_control() */
+	public function test_register_control_args() {
+
+		$setting_args = [
+			'name'        => 'Setting Name',
+			'description' => 'Setting Description',
+			'options'     => [ 'black', 'white' ],
+		];
+
+		$this->get_settings_instance()->register_setting( 'color', Setting::TYPE_STRING, $setting_args );
+
+		$this->get_settings_instance()->register_control( 'color', Control::TYPE_SELECT, [
+			'options' => [
+				'black' => 'Black',
+				'white' => 'White',
+				'red'	=> 'Red',
+			],
+		] );
+
+		$control = $this->get_settings_instance()->get_setting( 'color' )->get_control();
+
+		// TODO: uncomment assert for $control->get_options when https://github.com/skyverge/wc-plugin-framework/pull/453 is merged {WV 2020-03-20}
```

Comment:
```suggestion
```

Reply:
Actually, I think the PR number in your comment may be incorrect and you meant https://github.com/skyverge/wc-plugin-framework/pull/443. In that case, the lines should not be uncommented/removed yet.

Reply:
You're right... and I still committed all the suggestions before realizing we haven't merged that PR yet 🤦🏻‍♂️. I'm sorry for all the confusion.

I reverted the suggestion commit to comment the assertion again and be able to merge.